// ngx-age-validator/__ivy_ngcc__/fesm2015/ngx-age-validator.js
import * as moment from 'moment';
import { NgModule } from '@angular/core';

import * as ɵngcc0 from '@angular/core';
function NgxAgeValidator(min, max) {
    return (control) => {
        const birthday = control.value;
        // console.log("birthday", birthday);
        if (birthday !== undefined) {
            const years = moment().diff(birthday, 'years');
            // console.log("years", years);
            if (min && min > years) {
                return { 'tooJung': true };
            }
            if (max && max < years) {
                return { 'tooOld': true };
            }
            return null;
        }
        return { 'ageValidation': true };
    };
}

class NgxAgeValidatorModule {
}
NgxAgeValidatorModule.ɵfac = function NgxAgeValidatorModule_Factory(t) { return new (t || NgxAgeValidatorModule)(); };
NgxAgeValidatorModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgxAgeValidatorModule });
NgxAgeValidatorModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxAgeValidatorModule, [{
        type: NgModule,
        args: [{
                declarations: [],
                imports: [],
                exports: []
            }]
    }], null, null); })();

/*
 * Public API Surface of ngx-age-validator
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgxAgeValidator, NgxAgeValidatorModule };
